app/models/order_detail.rb

```diff
@@ -266,6 +265,11 @@ def can_be_viewed_by?(user)
       .where("dispute_at IS NULL OR dispute_resolved_at IS NOT NULL")
   }
 
+  scope :need_statement, lambda { |facility|
```

Comment:
What I'm trying to suggest is that I believe we can replace`.where("products.facility_id" => facility.id)` with `for_facility(facility)`, and that method should take care of it for us, then we don't need the new `need_any_statement` scope. Does that make sense? Do you think that might work?